pkg/sql/parser/sql.y

```diff
@@ -9492,6 +9492,10 @@ show_jobs_stmt:
   {
     $$.val = &tree.ShowChangefeedJobs{}
   }
+| SHOW CHANGEFEED JOBS WITH WATCHED_TABLES
```

Comment:
Please add parser tests like Blathers said in https://github.com/cockroachdb/cockroach/pull/151131#issuecomment-3140684105

`pkg/sql/parser/testdata/show` seems like the right file for them